framework/include/actions/ActionFactory.h

```diff
@@ -85,7 +85,7 @@ class ActionFactory
 
   std::string getTaskName(const std::string & action);
 
-  MooseSharedPointer<Action> create(const std::string & action, const std::string & parsing_syntax, InputParameters parameters);
+  MooseSharedPointer<Action> create(const std::string & action, const std::string & action_name, InputParameters parameters);
```

Comment:
I dug into your Parser change and determined that this isn't the `action_name`. You've basically peeled off the last part of the parsing syntax (i.e. `Kernels/diff`) so this would be "diff" in this case. Am I missing something?